contrib/nydusify/pkg/converter/provider/source.go

```diff
@@ -100,16 +101,27 @@ func (sl *defaultSourceLayer) Mount(ctx context.Context) ([]mount.Mount, func()
 	digestStr := sl.desc.Digest.String()
 
 	if err := utils.WithRetry(func() error {
-		// Pull the layer from source
-		reader, err := sl.remote.Pull(ctx, sl.desc, true)
-		if err != nil {
-			return errors.Wrap(err, fmt.Sprintf("Decompress source layer %s", digestStr))
+		pull := func(plainHTTP bool) error {
+			// Pull the layer from source
+			reader, err := sl.remote.PullWithHTTP(ctx, sl.desc, true, plainHTTP)
+			if err != nil {
+				return errors.Wrap(err, fmt.Sprintf("Decompress source layer %s", digestStr))
+			}
+			defer reader.Close()
+
+			// Decompress layer from source stream
+			if err := utils.UnpackTargz(ctx, sl.mountDir, reader); err != nil {
+				return errors.Wrap(err, fmt.Sprintf("Decompress source layer %s", digestStr))
+			}
+
+			return nil
 		}
-		defer reader.Close()
 
-		// Decompress layer from source stream
-		if err := utils.UnpackTargz(ctx, sl.mountDir, reader); err != nil {
-			return errors.Wrap(err, fmt.Sprintf("Decompress source layer %s", digestStr))
+		if err := pull(false); err != nil {
+			if !errutil.IsErrHTTPResponseToHTTPSClient(err) && !errutil.IsErrConnectionRefused(err) {
+				return err
+			}
+			return pull(true)
```

Comment:
So we always retry without tls. How does the `insecure-xxx` command line argument work?

Reply:
Updated, `--insecure xxx` means `Skip verifying server certs for HTTPS xxx registry`. We always fall back to HTTP if HTTPS fails (with a warning message).